src/tests/test_connection.py

```diff
@@ -1184,3 +1185,55 @@ def test_database_in_directory_that_doesnt_exist(tmp_empty, uri, expected):
         SQLAlchemyConnection(engine=create_engine(uri))
 
     assert expected in str(excinfo.value)
+
+
+query_expected_outputs = [
+    ("SELECT * FROM table", True),
+    ("SUMMARIZE table", True),
+    ("FROM table SELECT *", True),
+    ("UPDATE table SET column=value", False),
+    ("INSERT INTO table (column) VALUES (value)", False),
+    ("INSERT INTO table SELECT * FROM table2", False),
+    ("UPDATE table SET column=10 WHERE column IN (SELECT column FROM table2)", False),
+    ("WITH x AS (SELECT * FROM table) SELECT * FROM x", True),
+    ("WITH x AS (SELECT * FROM table) INSERT INTO y SELECT * FROM x", False),
+    ("", False),
+    ("DELETE FROM table", False),
+    ("WITH summarize AS (SELECT * FROM table) SELECT * FROM summarize", True),
+    (
+        "WITH summarize AS (SELECT * FROM table) INSERT INTO y SELECT * FROM summarize",
+        False,
+    ),
+    ("UPDATE table SET column='SELECT'", False),
+    ("CREATE TABLE SELECT (id INT)", False),
+    ("CREATE TABLE x (SELECT VARCHAR(100))", False),
+    ('INSTALL "x"', False),
+    ("SELECT SUM(column) FILTER (WHERE column > 10) FROM table", True),
+    ("SELECT column FROM (SELECT * FROM table WHERE column = 'SELECT') AS x", True),
+    # Invalid SQL returns false
+    ("INSERT INTO table (column) VALUES ('SELECT')", False),
+    pytest.param(
+        "SELECT FROM table WHERE (column = 'value'",
+        False,
+        marks=pytest.mark.xfail(
+            reason="sqlparse does not notice the missing close paren"
+        ),
+    ),
```

Comment:
we can remove this test case, it's ok. if the SQL is invalid, the SQL engine will just complain and raise an error

Reply:
[fixed](https://github.com/ploomber/jupysql/pull/895/files#diff-ecfdf2a1f3adc925d3c32b8aa52540474a10a4160d552f281d7dbd85a0260170R1221)